image/helpers.go

```diff
@@ -53,11 +46,79 @@ type Store interface {
 	Assertion(assertType *asserts.AssertionType, primaryKey []string, user *auth.UserState) (asserts.Assertion, error)
 }
 
+// ToolingStore wraps access to the store for tools.
+type ToolingStore struct {
+	sto  Store
+	user *auth.UserState
+}
+
+func newToolingStore(arch, storeID string) (*ToolingStore, error) {
+	cfg := store.DefaultConfig()
+	cfg.Architecture = arch
+	cfg.StoreID = storeID
+	var user *auth.UserState
+	if authFn := os.Getenv("UBUNTU_STORE_AUTH"); authFn != "" {
+		var err error
+		user, err = readAuthFile(authFn)
+		if err != nil {
+			return nil, err
+		}
+	}
+	sto := store.New(cfg, nil)
+	return &ToolingStore{
+		sto:  sto,
+		user: user,
+	}, nil
+}
+
+type authData struct {
```

Comment:
ended up using SNAPD_AUTH_DATA_FILENAME for in client  and UBUNTU_STORE_AUTH_DATA_FILENAME for this one